src/main/java/ui/UI.java

```diff
@@ -443,6 +448,27 @@ private void primaryRepoChanged(String repoId) {
         logic.openPrimaryRepository(repoId);
         logic.setDefaultRepo(repoId);
     }
+    
+    public void switchDefaultRepo(){
+        logger.info("Triggered switch default repo");
+        String[] openRepos = logic.getOpenRepositories().toArray(new String[0]);
```

Comment:
`RepositorySelector` already accesses the combo box's items through `getItems()` in its `loadContents` method. Try using `getItems()` to see if you can get the full repo list?